src/v/cluster/partition_balancer_planner.cc

```diff
@@ -163,6 +163,29 @@ class partition_balancer_planner::request_context {
     ss::abort_source& _as;
 };
 
+std::pair<uint64_t, uint64_t> partition_balancer_planner::get_node_bytes_info(
+  const node::local_state& node_state) {
+    const auto& state = node_state.log_data_size;
+    if (likely(state)) {
+        // It is okay to account reclaimable space as free space even through
+        // the space is not readily available. During the partition move if the
+        // disk is filled up, trimmer cleans up reclaimable space to make room
+        // for the move.
+        auto actual_free = std::max(
+          state.value().data_target_size - state.value().data_current_size,
+          0UL);
+        auto total_free = actual_free + state.value().data_reclaimable_size;
```

Comment:
I wonder if this should be capped to `data_target_size`

Reply:
Maybe @dotnwat can chime in with whether the data_reclaimable_size get surpass data_target_size. If so, total - free will be negative but used as a uint